### PR TITLE
Rename T-Mobile to Deutsche Telekom

### LINKED ISSUE #11782: Rename: T-Mobile
### Brand Name

T-Mobile

### Official Resources for Icon and Color

“[T-Mobile](https://en.wikipedia.org/wiki/T-Mobile_(brand))” is one of many brands used by Deutsche Telekom AG (German Telecom). They also operate under the names “Magenta” (Austria) or just “Telekom” (e.g. Germany, Hungary, Romania), where they use the same logo. Thus, it should be renamed to “Telekom” or “Deutsche Telekom”, but it should also be possible to find it by searching “T-Mobile” or perhaps even “Magenta”.

### Additional Comments

_No response_

### PR DESCRIPTION
**Pull request by DerBraky**
**Issue:** closes #11782

**Popularity metric:**
- Similarweb Global Rank: [2,753](https://www.similarweb.com/website/telekom.com) ([Company website](https://www.telekom.com/en))

### Checklist

- [x] I updated the JSON data in `_data/simple-icons.json`
- [ ] I optimized the icon with SVGO or SVGOMG
- [ ] The SVG `viewbox` is `0 0 24 24`

### Description

[T-Mobile](https://en.wikipedia.org/wiki/T-Mobile_(brand)) is only a brand for three mobile network operators (Czechia, Poland and the US) which are using the same logo/icon (the *T logo*) as its parent company, [Deutsche Telekom AG](https://en.wikipedia.org/wiki/Deutsche_Telekom). In [eight other European countries](https://en.wikipedia.org/wiki/Deutsche_Telekom#Operations), the names of its subsidiary mobile network operators include “Telekom”. Deutsche Telekom also provides other telecommunications and IT services in Germany and has another subsidiary called [T-Systems](https://en.wikipedia.org/wiki/T-Systems), which also uses the *T logo*. Thus, I think renaming it to “Telekom” would make the most sense.

“T-Mobile” will remain as an alias name, as well as “Deutsche Telekom”. For its Austrian subsidiary, I added a localized name, as they use a slightly different trading name ([Magenta Telekom](https://en.wikipedia.org/wiki/Magenta_Telekom)). For North Macedonia, I also added the Cyrillic version of their company name ([Македонски Телеком](https://en.wikipedia.org/wiki/Makedonski_Telekom)). Since the remaining subsidiary name only contain “Telekom” or “T-Mobile” next to their country name, I wouldn't include them there (e.g. [Hrvatski Telekom](https://en.wikipedia.org/wiki/Hrvatski_Telekom)).

Apart from renaming it, I haven't changed the icon itself.

For more context, see #11782

### CODE REVIEW

**Comment by uncenter** [comment]
I would be fine renaming it to Deutsche Telekom but why just "Telekom"? That has no meaning and isn't a brand or company.

**Comment by uncenter** [comment]
Let's rename it to "Deutsche Telekom", the others can be aliases. It's either that or leave it as T-Mobile, anything else seems unclear.

File: _data/simple-icons.json [hunk 1]
```diff
@@ -4240,6 +4240,22 @@
...
         },
+        {
+            "title": "Deutsche Telekom",
+            "hex": "E20074",
+            "source": "https://tmap.t-mobile.com/portals/pro74u7a/EXTBrandPortal",
+            "guidelines": "https://tmap.t-mobile.com/portals/pro74u7a/EXTBrandPortal",
+            "aliases": {
+                "aka": [
+                    "T-Mobile",
+                    "Telekom"
```
Comment (adamrusted) [inline _data/simple-icons.json:4251]: Alias isn't needed if the word is in the main title.

**Comment by LitoMore** [comment]
Converting to draft since it's a breaking change.

**Pull request merged**